src/api/spec/db/data/remove_duplicate_repositories_spec.rb

```diff
@@ -0,0 +1,77 @@
+require 'rails_helper'
+require Rails.root.join('db/data/20170306084550_remove_duplicate_repositories.rb')
+require Rails.root.join('db/migrate/20170306084558_change_repositories_remote_project_name_to_not_null.rb')
+
+RSpec.describe RemoveDuplicateRepositories, type: :migration do
+  # This migration does not allow NULL values for remote_project_name column
+  # We need this migration to not be run to create test data
+  # 20170306084558_change_repositories_remote_project_name_to_not_null.rb
+  let(:schema_migration_version) { 20_170_306_084_558 }
+  let(:schema_migration) { ChangeRepositoriesRemoteProjectNameToNotNull.new }
+  let(:data_migration_version) { 20_170_306_084_550 }
+  let(:data_migration) { RemoveDuplicateRepositories.new }
+
+  describe '.up' do
+    before do
+      schema_migration.down
+    end
+
+    after do
+      DataMigrate::DataSchemaMigration.find_or_create_by(version: data_migration_version.to_s)
+      schema_migration.up
+    end
+
+    let(:remote_project) { create(:remote_project) }
+    let!(:remote_repository) { create(:repository, project: remote_project, name: 'standard', remote_project_name: 'openSUSE.org:Foo') }
+    let!(:remote_repository_with_same_name) do
+      create(:repository, project: remote_project, name: 'standard', remote_project_name: 'openSUSE.org:Bar')
+    end
+
+    let(:project) { create(:project) }
+    let!(:repository) do
+      # we need to bypass the validation
```

Comment:
why?

Reply:
Because it is the main point of this migration and test: we want to generate invalid data which gets migrated in the data migration :smile: 
Should I add it to the comment?

Reply:
Yes, I think it would be useful :wink: